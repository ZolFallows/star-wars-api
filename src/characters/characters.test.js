import React from 'react';
// import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Characters from './characters';


describe(`Characters Component`, () => {
    it('renders empty when sections not supplied', () => {
        const wrapper = shallow(<Characters />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('only renders when character search triggers', () => {
      const wrapper = shallow(<Characters />);
      const input = wrapper.find('input');
      input.forEach(button => {
        button.simulate('onKeyPress')
      })
      expect(toJson(wrapper)).toMatchSnapshot();
    })
  })