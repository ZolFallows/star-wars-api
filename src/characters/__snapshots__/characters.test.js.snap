// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters Component only renders when character search triggers 1`] = `
<Fragment>
  <div
    className="search-form"
  >
    <input
      autoComplete="off"
      name="people"
      onKeyPress={[Function]}
      placeholder="Enter character name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="search-result"
  />
</Fragment>
`;

exports[`Characters Component renders empty when sections not supplied 1`] = `
<Fragment>
  <div
    className="search-form"
  >
    <input
      autoComplete="off"
      name="people"
      onKeyPress={[Function]}
      placeholder="Enter character name"
      required={true}
      type="text"
    />
  </div>
  <div
    className="search-result"
  />
</Fragment>
`;

exports[`Chareacters component renders the UI as expected 1`] = `
Array [
  <div
    className="search-form"
  >
    <input
      autoComplete="off"
      name="people"
      onKeyPress={[Function]}
      placeholder="Enter character name"
      required={true}
      type="text"
    />
  </div>,
  <div
    className="search-result"
  />,
]
`;

exports[`Chareacters component renders the UI as expected with no unreads 1`] = `
Array [
  <div
    className="search-form"
  >
    <input
      autoComplete="off"
      name="people"
      onKeyPress={[Function]}
      placeholder="Enter character name"
      required={true}
      type="text"
    />
  </div>,
  <div
    className="search-result"
  />,
]
`;
